### PR TITLE
To find maximum element in each contiguous subarray of size k in O(n) (C++) 

### PR DESCRIPTION
**Pull request by divya-rai-42**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [Yes ] I've read the contribution guidelines.
- [Yes ] I've checked the issue list before deciding what to submit.
- [Yes ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
This pull request is under the issue #4878

### CODE REVIEW

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -119,6 +119,7 @@
 - [Plus One Problem](cp/Plus_One.cpp)
 - [Minimum Cost to generate N Cells](cp/min_cost_to_generate_n_cells.cpp)
 - [Groups Anagrams](cp/Group_Anagrams.cpp)
+- [Maximum element in each subarray of size k](cp/ma_element_in_k_size_subarray.cpp)
```
Comment (atarax665) [inline C-Plus-Plus/README.md]:
```suggestion
- [Maximum element in each subarray of size k](cp/max_element_in_k_size_subarray.cpp)
```

File: C-Plus-Plus/cp/max_element_in_k_size_subarray.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
+/*This algorithm uses deque. Only the significant elements of the current window are stored in the deque in decreasing
```
Comment (atarax665) [inline C-Plus-Plus/cp/max_element_in_k_size_subarray.cpp]: Edit the description. The objective of the program is not defined clearly.

File: C-Plus-Plus/cp/max_element_in_k_size_subarray.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        max=dq.front();
+        cout<<arr[max]<<" ";
+    }
+    return;
+}
+//Driver Code
+int main()
+{
+    int n;
+    cin>>n;
```
Comment (atarax665) [inline C-Plus-Plus/cp/max_element_in_k_size_subarray.cpp:49]: Add input-output prompts to make the program more interactive.

File: C-Plus-Plus/cp/max_element_in_k_size_subarray.cpp [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    for(int i=k;i<n;i++)
+    {
+        /*Checking whether on moving to the next window, dq.front() still remains the member of the current window or not.
+          If not, then that element is removed from the current window*/
+        if(i-k==max)
+        dq.pop_front();
+        //For the element at ith position, the previous smaller element is insignificant. Thus, remove it
+        while(dq.empty()==false && arr[dq.back()]<=arr[i])
+        dq.pop_back();
+        dq.push_back(i);//Adding new element to the current window
```
Comment (atarax665) [inline C-Plus-Plus/cp/max_element_in_k_size_subarray.cpp]: Shift the comment above the line of code.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Resolve merge conflicts as well

**Comment by divya-rai-42** [comment]
Sir, I have done the desired changes. Please review it further.

**Comment by divya-rai-42** [comment]
Can you please help me out why I am getting merge conflicts? When I resolved the previous issues, there were no conflicts. But after sometime , it shows merge conflicts. Is it because other commits have been done to the master branch of the main repo after I resolved the conflicts?
Also please tell me what is meant by label 'outdated'?

**Comment by divya-rai-42** [comment]
Sir, please help me out to resolve the merge conflicts and answer some of my queries that I have asked earlier!

**Comment by divya-rai-42** [comment]
Sir please guide me if I need to do more changes!

**Comment by divya-rai-42** [comment]
@ankitaggarwal23  Please review it

File: C-Plus-Plus/cp/max_element_in_k_size_subarray.cpp [hunk 5]
```diff
@@ -0,0 +1,65 @@
+/*Through this algorithm, we can find the maximum element in each of the k sized subarray.
+The algorithm uses deque. Only the significant elements of the current window are stored in the deque in decreasing
+order. The elements which are lesser than any of the elements of the deque and are to the left of that particular element
+are insignificant. As the current window changes, significant values are updated. */
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/max_element_in_k_size_subarray.cpp]:
```
/*
Through this algorithm, we can find the maximum element in each of the k sized subarray.
The algorithm uses deque. Only the significant elements of the current window are stored in the deque in decreasing
order. The elements which are lesser than any of the elements of the deque and are to the left of that particular element
are insignificant. As the current window changes, significant values are updated.
 */
```
Rewrite like that

File: C-Plus-Plus/cp/max_element_in_k_size_subarray.cpp [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    func(arr,n,k);
+    return 0;
+}
+
+/*Time Complexity: O(n)
+  Space Complexity: O(n)
+  Sample Input: 10
+                5 6 7 4 1 2 5 8 9 3
+                4
+  Sample Output: 7 7 7 5 8 9 9 */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/max_element_in_k_size_subarray.cpp]: Write the outputs in the way your code is producing.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by divya-rai-42** [comment]
@ankitaggarwal23  sir I have done the changes. Please review it!

**Review by ankitaggarwal23 — COMMENTED** [review]
Don't push more than one code file in a single PR.

**Review by ankitaggarwal23 — COMMENTED** [review]
Open the new PR if you want this to be reviewed and add the proper files

**Comment by divya-rai-42** [comment]
@ankitaggarwal23  Sorry sir, but I had not attached two code files. It would be very grateful of you if you tell me which is the other file that I have added (maybe unknowingly) , so that I don't repeat this again. I am very new to github , so am unable to figure out which is the other file.

**Comment by ankitaggarwal23** [comment]
Minimum sum partition

**Pull request closed without merging**